### PR TITLE
should force load react-components which send over turbo-stream

### PR DESCRIPTION
**Pull request by theforestvn88**
### Summary

Support React On Rails w/Hotwire and Turbo Streams #1493, #1508

All turbo events (`turbo:before-render`, `turbo:render`,.. ) will not dispatch with `TurboStream`, except [turbo:before-stream-render](https://github.com/hotwired/turbo-rails/blob/main/app/assets/javascripts/turbo.js#L5071).
And it looks like `turbo-rails` did not support a `after-stream-render` event, i don't know why yet.

But `hotwired` send html, `react_on_rails` send `script`, right ?
So my solution: send a script that will trigger load react component along with it.

- I add a new `render option`: `force_load` which if true then the trigger load script will be concat with the component itself.
- about the trigger load script : i not use `reactOnRailsPageLoaded` which will query and reload all current react-components (not effective), i introduce new method `reactOnRailsComponentLoaded(react-component-dom-id)` to load directly the react-component that the script be sent along with.

### Other Information

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1614)

## Summary by CodeRabbit

- **New Features**
  - Added a new configuration option to control forced loading of components.
  - Introduced script inclusion based on rendering options.
  - Implemented Turbo Stream capabilities for React components.

- **Enhancements**
  - Enhanced component loading mechanisms to support dynamic DOM IDs.

- **Documentation**
  - Updated helper and integration test documentation to reflect new features and use cases.

- **Bug Fixes**
  - Adjusted configuration defaults and component pack loading settings.

- **Refactor**
  - Streamlined React component rendering and configuration retrieval processes.

- **Tests**
  - Added new test cases for React component interactions with Turbo Streams.

### CODE REVIEW

File: spec/dummy/app/views/pages/turbo_frame_tag_hello_world.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<%= turbo_frame_tag 'hello-turbo-stream' do %>
+    <%= button_to "send me hello-turbo-stream component", turbo_stream_send_hello_world_path %>
+<% end %>
```
Comment (justin808) [inline spec/dummy/app/views/pages/turbo_frame_tag_hello_world.html.erb:3]: please add return

**Review by justin808 — COMMENTED** [review]
@Judahmeek please review carefully

**Comment by justin808** [comment]
@theforestvn88 Thanks! @Judahmeek and I will review!

File: spec/dummy/app/views/pages/turbo_stream_send_hello_world.turbo_stream.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<%= turbo_stream.update 'hello-turbo-stream' do %>
+    <%= react_component("HelloTurboStream", props: @app_props_hello_from_turbo_stream, force_load: true) %>
+<% end %>
```
Comment (Judahmeek) [inline spec/dummy/app/views/pages/turbo_stream_send_hello_world.turbo_stream.erb:3]: please add return

File: spec/dummy/Gemfile [hunk 3]
```diff
@@ -5,3 +5,5 @@ source "https://rubygems.org"
 eval_gemfile File.expand_path("../../Gemfile.development_dependencies", __dir__)
 
 gem "react_on_rails", path: "../.."
+
+gem "turbo-rails"
```
Comment (Judahmeek) [inline spec/dummy/Gemfile]: please add this gem to the Gemfile.development_dependencies file instead of here.

**Review by Judahmeek — CHANGES_REQUESTED** [review]
Overall, this seems like a solid piece of work.

Please make the requested changes while I look into resolving the CI failures related to Bundler version mismatch (in other words, the CI failures not related to your PR)

**Comment by Judahmeek** [comment]
@theforestvn88 please rebase your PR on [shakacode:judahmeek/bundler-version](https://github.com/shakacode/react_on_rails/tree/judahmeek/bundler-version)

**Comment by justin808** [comment]
@theforestvn88 that PR is merged. Rebase on master please!

File: spec/dummy/app/views/pages/turbo_frame_tag_hello_world.html.erb (hunk 1 above)
Comment (theforestvn88) [reply spec/dummy/app/views/pages/turbo_frame_tag_hello_world.html.erb:3]: hi, i don't get it! what should i 'return' here. some document like other pages, right ?

**Comment by justin808** [comment]
> @theforestvn88 please rebase your PR on [shakacode:judahmeek/bundler-version](https://github.com/shakacode/react_on_rails/tree/judahmeek/bundler-version)

@Judahmeek that wasn't good advice as the branch got deleted.

**Comment by justin808** [comment]
@Judahmeek @theforestvn88 I recreated the branch so I could reopen this PR.

Can I see an example of this fix used in https://github.com/shakacode/react-webpack-rails-tutorial? @theforestvn88 can you make a PR against that repo?

**Comment by justin808** [comment]
Looks like #1620 takes over.

**Pull request closed without merging**